lib/ace/ext/papeeria/biblio_completer.coffee

```diff
@@ -0,0 +1,117 @@
+foo = null # ACE builder wants some meaningful JS code here to use ace.define instead of just define
+define((require, exports, module) ->
+  LatexParsingContext = require("ace/ext/papeeria/latex_parsing_context")
+  Autocomplete = require('ace/autocomplete')
+  util = require("ace/autocomplete/util")
+  lang = require("../../lib/lang")
+
+  class BiblioFilesCompleter extends Autocomplete.Autocomplete
+    constructor: ->
+      super()
+    setFiles: (files) => @files = files
+    _setCompletions: =>
+      matches = @files.map((file) -> return {
+        name: file.name
+        file: file
+        meta: file.path
+        completer: this
+      })
+      matches.push({
+        name: "Create new file"
+        meta: ""
+        completer: this
+      })
+      @completions = new Autocomplete.FilteredList(matches)
+
+    setCreateCallback: (callback) => @createCallback = callback
+    setInsertCallback: (callback) => @insertCallback = callback
+
+    _doInsert: (file) =>
+      @insertCallback?(file)
+      @detach()
+
+    _doCreate: =>
+      @createCallback?((=> @editor.completer = null), @_doInsert)
+
+    updateCompletions: (keepPopupPosition) =>
+      @_setCompletions()
+      prefix = util.getCompletionPrefix(@editor)
+      @openPopup(@editor, prefix, keepPopupPosition)
+
+    insertMatch: (editor, data) =>
+      if not data?
+        data = @popup.getData(@popup.getRow())
+      if not data?
+        @detach()
+        return
+
+      if data.name == "Create new file"
+        @_doCreate()
+      else
+        @_doInsert(data.file)
+
+    detach: =>
+      @editor.completer = new Autocomplete.Autocomplete()
```

Comment:
try adding `@isDisposable = true` field (like it's done in `spellchecker_popup.coffee`) and chances are you won't need to override `detach`